src/kaboodle.rs

```diff
@@ -718,6 +733,7 @@ impl KaboodleInner {
         self.maybe_broadcast_join().await;
         self.handle_suspected_peers().await;
         self.ping_random_peer().await;
+        self.handle_incoming_ping_requests().await;
```

Comment:
I've tried to clean this up a few times but it's beyond my current Rust abilities. I'd welcome any suggestions; I had tried something like this:

````rust
type TaskFn = fn(&mut KaboodleInner) -> impl Future<Output = ()>;
let tasks: Vec<TaskFn> = vec![
    Self::maybe_broadcast_join,
    Self::handle_suspected_peers,
    Self::ping_random_peer,
    Self::handle_incoming_ping_requests,
];
for task in tasks.iter() {
    task(&mut self).await;
}
````

However, this fails because `error[E0658]: `impl Trait` in type aliases is unstable` and I don't know what else to do. :sob: